data_validation/util.py

```diff
@@ -62,3 +64,53 @@ def dvt_config_string_to_dict(config_string: str) -> dict:
         raise exceptions.ValidationException(
             f"Invalid JSON format in connection parameter dictionary string: {config_string}"
         ) from exc
+
+
+def datetime_to_iso_format(obj):
+    """
+    Converts a datetime object, pandas Timestamp, or integer timestamp to an ISO 8601 formatted string.
+
+    Args:
+        obj (datetime.datetime or pandas.Timestamp or int): The object to convert.
+
+    Returns:
+        str: An ISO 8601 formatted string representing the datetime.
+
+    Raises:
+        TypeError: If the object is not a datetime, pandas Timestamp, or integer timestamp.
+
+    Examples:
+        >>> from datetime import datetime
+        >>> import pandas as pd
+        >>> datetime_to_iso(datetime(2023, 10, 26, 10, 30, 0))
+        '2023-10-26T10:30:00'
+
+        >>> datetime_to_iso(pd.Timestamp('2023-10-26 10:30:00'))
+        '2023-10-26T10:30:00'
+
+        >>> datetime_to_iso(1677609600)  # Seconds since epoch
+        '2023-03-01T00:00:00+00:00'
+
+        >>> datetime_to_iso(1677609600000) # milliseconds since epoch
+        '2023-03-01T00:00:00+00:00'
+
+        >>> datetime_to_iso("invalid")
+        Traceback (most recent call last):
+            ...
+        TypeError: Object of type str is not JSON serializable
+    """
+    if isinstance(obj, datetime):
+        return obj.isoformat()
+    elif isinstance(obj, PandasTimestamp):
```

Comment:
Seeing as the start and end times can only be datetime values (we control where they get set and it's always a datetime as far as I can see) it might be simpler to just use `.isoformat()` in place of calling `datetime_to_iso_format()`? I understand that you've tried to plan for future data types but perhaps we just cater for `PandasTimestamp` and `int` start/end times if we ever change how we set them in the first place and not add this unused code?